### ISSUE TITLE
VisibleDeprecationWarnings in test_given_colors_levels_and_extends

### ISSUE THREAD

**Issue by QuLogic** [matplotlib/matplotlib#7334]
On Travis, the `test_given_colors_levels_and_extends` test is generating two `VisibleDeprecationWarning`s:

```
.../matplotlib/colorbar.py:539: VisibleDeprecationWarning:
 boolean index did not match indexed array along dimension 0; dimension is 5 but corresponding boolean dimension is 4
  colors = np.asarray(colors)[igood]
.../matplotlib/colorbar.py:539: VisibleDeprecationWarning:
 boolean index did not match indexed array along dimension 0; dimension is 5 but corresponding boolean dimension is 3
  colors = np.asarray(colors)[igood]
```

The first occurs on the axes that is not filled and `extend='min'`; the second with unfilled and `extend='both'`. The passed colour list is 5 elements long and the levels list is 4 or 3. The warning can easily be fixed by trimming the colour list, but I'm not sure exactly where that should be.

I guess I'm not entirely sure why one less level is passed for `extend='both'` when only unfilled. cc @pelson who wrote the test

**Comment by pelson** [comment]
I'm sure it is developer error, rather than anything meaningful. Looks like there was an oversight of "both" for filled in https://github.com/matplotlib/matplotlib/blob/07adcdeb367c008229daafa932397c735123cd2d/lib/matplotlib/tests/test_contour.py#L199, and for min/max in non-filled at https://github.com/matplotlib/matplotlib/blob/07adcdeb367c008229daafa932397c735123cd2d/lib/matplotlib/tests/test_contour.py#L203.

**Comment by NelleV** [comment]
@pelson can you open a PR fixing that? :D

**Comment by pelson** [comment]
> @pelson can you open a PR fixing that? :D

Realistically, unfortunately not at the moment 😞 .

**Comment by NelleV** [comment]
:(

**Comment by pelson** [comment]
I know. Sad face from me too... 

![screen shot 2016-11-23 at 07 03 05](https://cloud.githubusercontent.com/assets/810663/20610506/e287d5f4-b2ee-11e6-9b97-e90257caf65b.png)

I'm sure it will turn around - it is just the growth of conda-forge that has swamped my time 😉

**Comment by QuLogic** [comment]
On further inspection, I'm not sure whether this is solely an error in the test. Removing colours from the input list _still_produces the same warning, so something in the contour itself is producing the wrong number of colours to be passed to the colorbar.

**Comment by dstansby** [comment]
Doesn't look like this is happening on current master any more: https://travis-ci.org/matplotlib/matplotlib/jobs/212484661

**Comment by QuLogic** [comment]
Pytest doesn't show warnings by default.

**Comment by dstansby** [comment]
Closed by #8737
